File: elemi_programozasi_tetelek/csoportositas/csoportositas.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

unordered_map<char, vector<string>> Csoportosit(vector<string>& lista)
{
    unordered_map<char, vector<string>> szotar;
    for (string& elem : lista)
    {
        szotar[elem[0]].push_back(elem);
    }
    return szotar;
}


int main()
{
    vector<string> szavak = { "repa", "retek", "mogyoro", "koran", "reggel", "ritkan", "rikkant", "a", "rigo" };

    unordered_map<char, vector<string>> szotar = Csoportosit(szavak);
    for (auto& par : szotar)
    {
        cout << par.first << ": ";
        for (string& elem : par.second)
        {
            cout << elem << " ";
        }
        cout << endl;
    }
}
    

File: elemi_programozasi_tetelek_rekurzivan/csoportositas/csoportositas.cs
using System;
using System.Collections.Generic;

namespace ConsoleApp1
{
    internal class Program
    {
        static Dictionary<char, List<string>> Csoportosit(List<string> lista)
        {
            Dictionary<char, List<string>> szotar = new Dictionary<char, List<string>>();
            foreach (string elem in lista)
            {
                char kulcs = elem[0];
                if (szotar.Keys.Contains(kulcs))
                {
                    szotar[kulcs].Add(elem);
                }
                else
                {
                    szotar[kulcs] = new List<string> { elem };
                }
            }
            return szotar;
        }

        static void Main(string[] args)
        {
            List<string> szavak = new List<string> { "répa", "retek", "mogyoró", "korán", "reggel", "ritkán", "rikkant", "a", "rigó"};

            Dictionary<char, List<string>> szotar = Csoportosit(szavak);
            foreach (char kulcs in szotar.Keys)
            {
                Console.Write(kulcs);
                Console.Write(": ");
                Console.WriteLine(string.Join(", ", szotar[kulcs]));
            }
        }
    }
}
